src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -89,6 +91,37 @@ void HSHomeObject::init_homestore() {
     LOGI("Initialize and start HomeStore is successfully");
 }
 
+void HSHomeObject::init_timer_thread() {
+    struct Context {
+        std::condition_variable cv;
+        std::mutex mtx;
+        bool created{false};
+    };
+    auto ctx = std::make_shared< Context >();
```

Comment:
This can just live on the stack right? We ensure it will out-live the lambda with the `wait` call immediately following. It's already being captured by reference.